libs/sdk-ui-dashboard/src/presentation/widget/common/errorUtils.ts

```diff
@@ -0,0 +1,54 @@
+// (C) 2021 GoodData Corporation
+import {
+    GoodDataSdkError,
+    isBadRequest,
+    isDataTooLargeToCompute,
+    isDataTooLargeToDisplay,
+    isNegativeValues,
+    isNoDataSdkError,
+    isProtectedReport,
+    isUnknownSdkError,
+} from "@gooddata/sdk-ui";
+import { isEmptyAfm } from "@gooddata/sdk-ui-ext/dist/internal";
+
+const nonExportableErrorGuards: Array<(item: unknown) => boolean> = [
+    isUnknownSdkError,
+    isBadRequest,
+    isNoDataSdkError,
+    isProtectedReport,
+    isDataTooLargeToCompute,
+    isEmptyAfm,
+];
+
+/**
+ * @internal
+ */
+export const isExportableError = (error: GoodDataSdkError | undefined): boolean => {
+    if (!error) {
+        return true;
+    }
+
+    return !nonExportableErrorGuards.some((guard) => guard(error));
+};
+
+const dataErrorGuards: Array<(item: unknown) => boolean> = [
+    isUnknownSdkError,
+    isBadRequest,
+    isNoDataSdkError,
+    isDataTooLargeToCompute,
+    isDataTooLargeToDisplay,
+    isProtectedReport,
+    isEmptyAfm,
+    isNegativeValues,
+];
+
+/**
+ * @internal
+ */
+export const isDataError = (error: GoodDataSdkError | undefined): boolean => {
```

Comment:
Is posible create union type of DataError ??? or use combineGuards from utils to get ReturnType ? 
And than set return valu as proper guard? 

export const isDataError = (error: GoodDataSdkError | undefined): error is DataErrorType =>

Reply:
updated